text/0000-modifiers.md

```diff
@@ -0,0 +1,242 @@
+
+Element modifiers allow for stable access of the DOM node they are installed on. This allows for programatic assess to DOM in Glimmer templates and also offers a more targeted construct for cases where classic components were being used. The introduction of this API will likely result in the proliferation of one or several popular addons for managing element event listeners, style and animation.
+
+## Detailed design
+
+### Invocation
+
+An element modifier is invoked in "element space". This is the space between `<` and `>` opening an HTML tag. For example:
+
+```hbs
+<button {{flummux}}></button>
+<span {{whipperwill 'carrot'}}><i>Some DOM</i></span>
+<b {{crum bing='whoop'}} zip="bango">Hm...</b>
+```
+
+Element modifiers may be invoked with params or hash arguments.
+
+### Definition and lookup
+
+A basic element modifier is defined with the type of `element-modifier`. For example these paths would be global element modifiers in an application:
+
+```
+Classic paths:
+
+  app/element-modifiers/flummux.js
+  app/element-modifiers/whipperwill.js
+
+MU paths:
+
+  src/ui/components/flummux/element-modifier.js
+  src/ui/routes/posts/-components/whipperwill/element-modifier.js
+```
+
+Element modifiers, like component and helpers, are eligible for local lookup. For example:
+
+```
+MU paths:
+
+  src/ui/routes/posts/-components/post-editor/flummux/element-modifier.js
+```
+
+The element modifier class is a default export from these files. For example:
+
+```js
+import Modifier from '@ember/modifier';
+
+export default class extends Modifier {}
+```
+
+### Lifecycle Hooks
+
+During rendering and teardown of a target element, any attached element modifiers will execute a series of hooks. These hooks are:
+
+* `didInsertElement`
+* `didUpdate`
+* `willDestroyElement`
+
+It is important to note that in server-side rendering environments none of the lifecycle events are called.
+
+#### `didInsertElement` semantics
+
+`didInsertElement` is called when the element is in the DOM and it's children are attached. The element is available on `this.element` of the modifier instance. Unlike classic components, the `didInsertElement` hook receives the positional and named params as arguments in the same way helpers do. This hook is only called once.
+
+This hook has the following timing semantics:
+
+**Always**
+- called **after** any children modifiers `didInsertElement` hook are called
+- called **after** DOM insertion
+
+ **May or May Not**
+- be called in the same tick as DOM insertion
+- have the the parent's children fully initialized in DOM
```

Comment:
Overall very positive to this RFC!

Minor thing: I guess "the parent's children" is the same as "the elements siblings"? If that's the case, would the term sibling be better?

Reply:
I'd also like clarification. If `children` is sibling elements, can't those belong to sibling components?